cmd/podman/common/create_opts.go

```diff
@@ -346,16 +347,25 @@ func ContainerCreateToContainerCLIOpts(cc handlers.CreateContainerConfig) (*Cont
 		Systemd:          "true", // podman default
 		TmpFS:            stringMaptoArray(cc.HostConfig.Tmpfs),
 		TTY:              cc.Config.Tty,
-		//Ulimit:            cc.HostConfig.Ulimits,            // ask dan, no documented format
-		Ulimit:      []string{"nproc=4194304:4194304"},
-		User:        cc.Config.User,
-		UserNS:      string(cc.HostConfig.UsernsMode),
-		UTS:         string(cc.HostConfig.UTSMode),
-		Mount:       mounts,
-		Volume:      volumes,
-		VolumesFrom: cc.HostConfig.VolumesFrom,
-		Workdir:     cc.Config.WorkingDir,
-		Net:         &netInfo,
+		User:             cc.Config.User,
+		UserNS:           string(cc.HostConfig.UsernsMode),
+		UTS:              string(cc.HostConfig.UTSMode),
+		Mount:            mounts,
+		Volume:           volumes,
+		VolumesFrom:      cc.HostConfig.VolumesFrom,
+		Workdir:          cc.Config.WorkingDir,
+		Net:              &netInfo,
+	}
+	if !rootless.IsRootless() {
```

Comment:
All of this should probably be happening in `pkg/specgen/generate/validate.go` in the resource validation block there.

Reply:
i might agree that the defaults part could ... but not the actual add

Reply:
No, we should be discarding even user-set values there, and adding warnings.